var/spack/repos/builtin/packages/arrow/package.py

```diff
@@ -50,7 +50,7 @@ class Arrow(CMakePackage, CudaPackage):
     depends_on("ninja", type="build")
     depends_on("openssl", when="+gandiva @6.0.0:")
     depends_on("openssl", when="@4.0.0:")
-    depends_on("orc", when="+orc")
+    depends_on("orc@1.6.11:", when="+orc")
```

Comment:
```
PKG-DIRECTIVES: 1 issue found
1. arrow: dependency on orc@1.6.11: cannot be satisfied by known versions of orc
    happening in /home/runner/work/spack/spack/var/spack/repos/builtin/packages/arrow/package.py
    known versions of orc are 1.6.5
```

Reply:
See https://github.com/spack/spack/pull/43222, maybe easier to combine in one PR.

Reply:
@payerle JFYI, see above

Reply:
OK, I was not sure what best practice was when dealing with two basically disjoint modifications to two packages.

I will bear in mind in the future.  I have added the changes for this PR into PR #43222 as that seems to be easiest way to merge the two.